rust_src/src/buffers.rs

```diff
@@ -383,6 +384,45 @@ impl LispBufferRef {
         let pos = buffer_bytes.add(offset) as *mut LispObject;
         *pos = value;
     }
+
+    pub fn reset(&mut self) {
+        let curr_buff = ThreadState::current_buffer();
+        let null_curr_buff = ThreadState::is_current_buffer_null();
```

Comment:
This feels like a C style solution. Perhaps the `current_buffer()` needs to return an `Option`?